src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -0,0 +1,321 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Reflection;
+
+internal static partial class Interop
+{
+    internal static partial class procfs
+    {
+        internal const string RootPath = "/proc/";
+        internal const string ProcUptimeFilePath = RootPath + "uptime";
+        private const int ExpectedStatFileParts = 44;
+
+#if DEBUG
+        static procfs()
+        {
+            var fields = typeof(ParsedStat).GetTypeInfo().DeclaredFields;
+            int fieldCount = 0;
+            foreach (var item in fields)
+            {
+                fieldCount++;
+            }
+            Debug.Assert(ExpectedStatFileParts == fieldCount);
+        }
+#endif
+
+        internal struct ParsedStat
+        {
+            internal int pid;
+            internal string comm;
+            internal char state;
+            internal int ppid;
+            internal int pgrp;
+            internal int session;
+            internal int tty_nr;
+            internal int tpgid;
+            internal uint flags;
+            internal ulong minflt;
+            internal ulong cminflt;
+            internal ulong majflt;
+            internal ulong cmajflt;
+            internal ulong utime;
+            internal ulong stime;
+            internal long cutime;
+            internal long cstime;
+            internal long priority;
+            internal long nice;
+            internal long num_threads;
+            internal long itrealvalue;
+            internal ulong starttime;
+            internal ulong vsize;
+            internal long rss;
+            internal ulong rsslim;
+            internal ulong startcode;
+            internal ulong endcode;
+            internal ulong startstack;
+            internal ulong kstkesp;
+            internal ulong kstkeip;
+            internal ulong signal;
+            internal ulong blocked;
+            internal ulong sigignore;
+            internal ulong sigcatch;
+            internal ulong wchan;
+            internal ulong nswap;
+            internal ulong cnswap;
+            internal int exit_signal;
+            internal int processor;
+            internal uint rt_priority;
+            internal uint policy;
+            internal ulong delayacct_blkio_ticks;
+            internal ulong guest_time;
+            internal long cguest_time;
+        }
+
+        private static string GetStatFilePathForProcess(int pid)
+        {
+            return RootPath + pid.ToString(CultureInfo.InvariantCulture) + "/stat";
```

Comment:
Do you want to constify "/stat" since you use it in a few places?  At that point maybe you do "/task/" as well?